src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -694,6 +695,43 @@ public function addProductAction(int $orderId, Request $request): Response
         ]);
     }
 
+    /**
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))", redirectRoute="admin_orders_index")
+     *
+     * @param int $orderId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function getProductPricesAction(int $orderId, Request $request): Response
+    {
+        try {
+            $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+            $productsForViewing = $orderForViewing->getProducts();
+            $productList = $productsForViewing->getProducts();
+
+            $result = [];
+            foreach ($productList as $product) {
+                $result[] = [
+                    'orderDetailId' => $product->getOrderDetailId(),
+                    'unitPrice' => $product->getUnitPrice(),
+                    'unitPriceTaxExclRaw' => $product->getUnitPriceTaxExclRaw(),
+                    'unitPriceTaxInclRaw' => $product->getUnitPriceTaxInclRaw(),
+                    'quantity' => $product->getQuantity(),
+                    'availableQuantity' => $product->getAvailableQuantity(),
+                    'totalPrice' => $product->getTotalPrice(),
+                ];
+            }
+        } catch (Exception $e) {
+            return $this->json(
+                ['message' => $this->getErrorMessageForException($e, $this->getErrorMessages($e))],
+                Response::HTTP_BAD_REQUEST
```

Comment:
Why using bad request?

Reply:
Mmm, good question @matthieu-rolland added this one
But why would you suggest instead? `Response::HTTP_INTERNAL_SERVER_ERROR` ?

Reply:
Depending on the part, if inputs from users are bad => bad request, if something went wrong from the server => internal server

Reply:
Hm, yes actually, we should return a 500 error, `Response::HTTP_INTERNAL_SERVER_ERROR` seems good.